keyboards/planck/keymaps/mgalisa/keymap.c

```diff
@@ -0,0 +1,400 @@
+}
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+  switch (keycode) {
+    case QWERTY:
+      if (record->event.pressed) {
+        print("mode just switched to qwerty and this is a huge string\n");
+        set_single_persistent_default_layer(_QWERTY);
+      }
+      return false;
+      break;
+    case COLEMAK:
+      if (record->event.pressed) {
+        set_single_persistent_default_layer(_COLEMAK);
+      }
+      return false;
+      break;
+    case DVORAK:
+      if (record->event.pressed) {
+        set_single_persistent_default_layer(_DVORAK);
+      }
+      return false;
+      break;
+    case BACKLIT:
+      if (record->event.pressed) {
+        register_code(KC_RSFT);
+        #ifdef BACKLIGHT_ENABLE
+          backlight_step();
+        #endif
+        #ifdef KEYBOARD_planck_rev5
+          PORTE &= ~(1<<6);
+        #endif
+      } else {
+        unregister_code(KC_RSFT);
+        #ifdef KEYBOARD_planck_rev5
+          PORTE |= (1<<6);
+        #endif
+      }
+      return false;
+      break;
+    case PLOVER:
+      if (record->event.pressed) {
+        #ifdef AUDIO_ENABLE
+          stop_all_notes();
+          PLAY_SONG(plover_song);
+        #endif
+        layer_off(_RAISE);
+        layer_off(_LOWER);
+        layer_off(_ADJUST);
+        layer_on(_PLOVER);
+        if (!eeconfig_is_enabled()) {
+            eeconfig_init();
+        }
+        keymap_config.raw = eeconfig_read_keymap();
+        keymap_config.nkro = 1;
+        eeconfig_update_keymap(keymap_config.raw);
+      }
+      return false;
+      break;
+    case EXT_PLV:
+      if (record->event.pressed) {
+        #ifdef AUDIO_ENABLE
+          PLAY_SONG(plover_gb_song);
+        #endif
+        layer_off(_PLOVER);
+      }
+      return false;
+      break;
+    case AUTO_SQ:
+      if (record->event.pressed) {
+        SEND_STRING("[]SS_TAP(X_LEFT)");
+      }
+      return false;
+      break;
+    case AUTO_CB:
+      if (record->event.pressed) {
+        SEND_STRING("{}SS_TAP(X_LEFT)");
+      }
+      return false;
+      break;
+    case AUTO_PA:
+      if (record->event.pressed) {
+        SEND_STRING("()SS_TAP(X_LEFT)");
+      }
+      return false;
+      break;
+    case SHRUG:
+      if (record->event.pressed) {
+        SEND_STRING("/shrugSS_TAP(X_ENT)");
+      }
+      return false;
+      break;
+  }
+  return true;
+}
+
+bool muse_mode = false;
+uint8_t last_muse_note = 0;
+uint16_t muse_counter = 0;
+uint8_t muse_offset = 70;
+uint16_t muse_tempo = 50;
+
+void encoder_update(bool clockwise) {
+  if (muse_mode) {
+    if (IS_LAYER_ON(_RAISE)) {
+      if (clockwise) {
+        muse_offset++;
+      } else {
+        muse_offset--;
+      }
+    } else {
+      if (clockwise) {
+        muse_tempo+=1;
+      } else {
+        muse_tempo-=1;
+      }
+    }
+  } else {
+    if (clockwise) {
+      #ifdef MOUSEKEY_ENABLE
+        register_code(KC_MS_WH_DOWN);
```

Comment:
You can use `tap_code(KC_MS_WH_DOWN)` etc. for these.

Reply:
Thanks - that piece is from where I copied the default Planck keymap (I haven't actually touched that piece on my own yet.) Is one better than/preferred over the other? If so I could submit a second PR to update the default.

Reply:
either works fine, actually.  But, specifically, `tap_code` calls register_code, adds a delay (if defined), and then unregister.  So, it does "mostly" the same thing. 

Also, you don't need to open an additional PR.  Any additional commits on the branch that you opened the PR from (`matt_dev` in this case) will automatically update this PR.

Reply:
Makes sense - I tested replacing `register_code()`/`unregister_code()` on my own keymap where it makes sense to do so and all seems well. Latest commit includes the changes in `planck/keymaps/default/keymap.c` as well.